compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/res/ComposeResources.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.res
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.toComposeImageBitmap
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.skia.Image
+
+@Composable
+fun Resource.rememberImageBitmap(): ImageBitmap = remember(this) { //todo move to commonMain
+    runBlocking {
+        Image.makeFromEncoded(readBytes()).toComposeImageBitmap()
+    }
+}
+
+@Composable
+fun Resource.rememberImageBitmapAsync(): State<ImageBitmap?> = remember(this) { //todo move to commonMain
```

Comment:
Should we return State instead of just `ImageBitmap?`?

The usage is inconvenient if we return `State`. In the [componentes.resources ](https://github.com/JetBrains/compose-jb/blob/7bfe7df5d07ec4714ccaf82ebe83d4967ba9062a/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/LoadState.kt#L111) we return `ImageBitmap`

Reply:
Yeah, good suggestion!
I will try to change to simple `ImageBitmap?` with ability to recompose with new value.